connector/failoverconnector/internal/state/pipeline_selector.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package state // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/failoverconnector/internal/state"
+
+import (
+	"sync"
+)
+
+// PipelineSelector is meant to serve as the source of truth for the target priority level
+type PipelineSelector struct {
+	CurrentIndex    int
+	StableIndex     int
```

Comment:
@djaglowski I think part of the confusion is my naming, so my intent with currentIndex is for it to serve as the source of truth for which pipeline we are currently on/ where the data is being exported to. It can be equal to either the stable index (which is the index of the highest priority pipeline that we know is working) or during a retry it will be temporarily equal to one of the retry index values

I don't want to name it retry index because when the current set pipeline is the stable one, it is equal to the stable index , not a retry index value. 

Maybe for clarity SetToRetryIndex should be renamed to SetToCurrentIndex, because what it is doing is setting the current index to the value that is passed.

Let me know if this is clear or if you see any issues with the concept.

Reply:
> I think part of the confusion is my naming, so my intent with currentIndex is for it to serve as the source of truth for which pipeline we are currently on/ where the data is being exported to. It can be equal to either the stable index (which is the index of the highest priority pipeline that we know is working) or during a retry it will be temporarily equal to one of the retry index values

It might come down to just naming, but what is the scenario which requires us to have a source of truth? Why isn't it enough to know the stable pipeline, and when retrying to know which pipeline we are retrying?

Reply:
@djaglowski well I guess because the consumer itself is abstracted from whether its in retry or not, it determines the nextConsumer it should be using by calling failover.getCurrentConsumer

Reply:
It's fine to keep it. I don't want to bog this down, just not super clear to me what it means. Let me know when this is ready for another review.

Reply:
It’s ready for another review. 

And gotcha, long story short it’s literally just telling ConsumeTraces which index to use when getting the next consumer. I think it’s necessary.